service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/ERLookupController.java

```diff
@@ -0,0 +1,90 @@
+
+package com.ericsson.eiffel.remrem.generate.controller;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map.Entry;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+public class ERLookupController {
+    private static RestTemplate restTemplate = new RestTemplate();
+
+    @Autowired
+    public void setRestTemplate(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public static ArrayList<String> ErlookupJson(JsonObject bodyJson) {
+        String[] ids = null;
+        ArrayList<String> aList = new ArrayList<String>();
+        JsonArray links = bodyJson.get("eventParams").getAsJsonObject().get("links").getAsJsonArray();
+        String s = links.toString();
+        JsonParser parser = new JsonParser();
+        JsonArray elemArray = parser.parse(s).getAsJsonArray();
+        for (int i = 0; i < elemArray.size(); i++) {
+            JsonObject json_obj = elemArray.get(i).getAsJsonObject();
+            if (json_obj.toString().contains("%lookup%")) {
+                String type = json_obj.get("type").getAsString();
+                links.remove(json_obj);
+                ids = Erlookup(json_obj);
+                aList.addAll(Arrays.asList(ids));
+                for (int j = 0; j < ids.length; j++) {
+                    JsonObject json_add = new JsonObject();
+                    json_add.addProperty("type", type);
+                    json_add.addProperty("target", ids[j]);
+                    links.add(json_add);
+                }
+            }
+        }
+        return aList;
+    }
+
+    public static String[] Erlookup(JsonObject jsonobj) {
```

Comment:
What is the difference between this and previous function? both takes JsonObject as input and one is named with "..Json" and the other is not named with "...Json". Write Javadoc for the functions.

Reply:
I will change function names and update the javadoc.

Reply:
ok.